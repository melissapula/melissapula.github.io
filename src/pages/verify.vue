<template>
<mdb-container>
  <mdb-row class="justify-content-center">
    <mdb-card class="card-body mt-4">
      <mdb-card-title>Calculate</mdb-card-title>
      <mdb-card-body>
        <mdb-row>
          <mdb-col col="3">
            <mdb-input type="number" outline placeholder="First number" v-model="firstNum"></mdb-input>
          </mdb-col>
          <mdb-col col="3">
            <select class="browser-default custom-select mt-4" v-model="operator">
              <option value="">Select Op</option>
              <option value="+">+</option>
              <option value="-">-</option>
              <option value="*">*</option>
              <option value="/">/</option>
            </select>
          </mdb-col>
          <mdb-col col="3">
            <mdb-input type="number" outline placeholder="Second number" v-model="secondNum"></mdb-input>
          </mdb-col>
          <mdb-col col="3" class="mt-2">
            <mdb-btn color="success" @click.native="calculate">Calculate</mdb-btn>
          </mdb-col>
        </mdb-row>
        <mdb-row>{{firstNum}} {{operator}} {{secondNum}} = {{result}}</mdb-row>
      </mdb-card-body>
    </mdb-card>
  </mdb-row>
</mdb-container>
</template>

<script>
export default {
  name: "verify",
  data() {
    return {
      firstNum: 0,
      secondNum: 0,
      operator: '',
      result: 0
    }
  },
  methods: {
    calculate() {
      if(this.firstNum && this.secondNum && this.operator !== '') {
        switch(this.operator) {
          case "+":
            this.result = parseInt(this.firstNum) + parseInt(this.secondNum);
            break
          case "-":
            this.result = parseInt(this.firstNum) - parseInt(this.secondNum);
            break
          case "*":
            this.result = parseInt(this.firstNum) * parseInt(this.secondNum);
            break
          case "/":
            this.result = parseInt(this.firstNum) / parseInt(this.secondNum);
            break
        }
      }
    }
  }
}
</script>

<style scoped>

</style>