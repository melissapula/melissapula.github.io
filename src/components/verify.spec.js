import {mount, createLocalVue} from '@vue/test-utils'
import verify from '../pages/verify'

test('mount a vue component', () => {
    const wrapper = mount(verify)
    console.log(wrapper)
})

describe('Verify.vue', () => {
    test('setup correctly', () => {
        expect(true).toBe(true)
    })
})

describe('verify', () => {
    it('has data', () => {
        expect(typeof verify.data).toBe('function')
    })
})

describe('Mounted Verify', () => {
    const wrapper = mount(verify);
    test('does a wrapper exist', () => {
        expect(wrapper.exists()).toBe(true)
    })
})

it('template contains markup', () => {
    const wrapper = mount(verify);
    expect(wrapper.html()).toContain('Calculate')
})

it('contains button', () => {
    const wrapper = mount(verify);
    expect(wrapper.find('mdb-btn').exists()).toBe(true)
})

it('renders correctly with different data', async () => {
    const wrapper = mount(verify);
    wrapper.setData({ firstNum: 5, secondNum: 10, operator: '+', result: 15 })
    await wrapper.vm.$nextTick()
    expect(wrapper.text()).toContain('5 + 10 = 15')
})

it('button click works', async () => {
    const wrapper = mount(verify);
    const button = wrapper.find('mdb-btn')
    await button.trigger('click')
})

it('adds correctly', () => {
    const wrapper = mount(verify);
    const localVue = createLocalVue()
    wrapper.setData({ localVue, firstNum: 10, secondNum: 2, operator: '+' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(12)
    wrapper.setData({ localVue, firstNum: 25, secondNum: 5, operator: '+' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(30)
})

it('subtracts correctly', () => {
    const wrapper = mount(verify);
    const localVue = createLocalVue()
    wrapper.setData({ localVue, firstNum: 13, secondNum: 4, operator: '-' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(9)
    wrapper.setData({ localVue, firstNum: 47, secondNum: 40, operator: '-' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(7)
})

it('multiplies correctly', () => {
    const wrapper = mount(verify);
    const localVue = createLocalVue()
    wrapper.setData({ localVue, firstNum: 5, secondNum: 7, operator: '*' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(35)
    wrapper.setData({ localVue, firstNum: 14, secondNum: 3, operator: '*' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(42)
})

it('divides correctly', () => {
    const wrapper = mount(verify);
    const localVue = createLocalVue()
    wrapper.setData({ localVue, firstNum: 15, secondNum: 3, operator: '/' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(5)
    wrapper.setData({ localVue, firstNum: 42, secondNum: 7, operator: '/' })
    wrapper.vm.calculate()
    expect(wrapper.vm.result).toBe(6)
})

