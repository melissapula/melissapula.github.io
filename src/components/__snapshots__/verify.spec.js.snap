// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount a vue component 1`] = `
<mdb-container firstnum="2">
  <mdb-row class="justify-content-center">
    <mdb-card class="card-body mt-4">
      <mdb-card-title>Calculate</mdb-card-title>
      <mdb-card-body>
        <mdb-row>
          <mdb-col col="3">
            <mdb-input type="number" outline="" placeholder="First number"></mdb-input>
          </mdb-col>
          <mdb-col col="3"><select class="browser-default custom-select mt-4">
              <option value="">Select Op</option>
              <option value="+">+</option>
              <option value="-">-</option>
              <option value="*">*</option>
              <option value="/">/</option>
            </select></mdb-col>
          <mdb-col col="3">
            <mdb-input type="number" outline="" placeholder="Second number"></mdb-input>
          </mdb-col>
          <mdb-col col="3" class="mt-2">
            <mdb-btn color="success">Calculate</mdb-btn>
          </mdb-col>
        </mdb-row>
        <!---->
      </mdb-card-body>
    </mdb-card>
  </mdb-row>
</mdb-container>
`;
